generate/utils.go

```diff
@@ -0,0 +1,1147 @@
+			addedCapsMap[cp] = true
+		}
+	}
+
+	g.spec.Process.Capabilities = finalCapList
+	return nil
+}
+
+// DropProcessCapabilities drop dropCaps from g.spec.Process.Capabilities.
+func (g Generator) DropProcessCapabilities(dropCaps []string) error {
+	capMappings := make(map[string]bool)
+	for _, cap := range capability.List() {
+		key := strings.ToUpper(cap.String())
+		capMappings[key] = true
+	}
+
+	dropCapsMap := make(map[string]bool)
+	for _, c := range dropCaps {
+		if !capMappings[c] {
+			return fmt.Errorf("Invalid value passed for dropping capability")
+		}
+		cp := fmt.Sprintf("CAP_%s", c)
+		dropCapsMap[cp] = true
+	}
+
+	var finalCapList []string
+	for _, c := range g.spec.Process.Capabilities {
+		if !dropCapsMap[c] {
+			finalCapList = append(finalCapList, c)
+		}
+	}
+	g.spec.Process.Capabilities = finalCapList
+	return nil
+}
+
+func mapStrToNamespace(ns string, path string) (rspec.Namespace, error) {
+	switch ns {
+	case "network":
+		return rspec.Namespace{Type: rspec.NetworkNamespace, Path: path}, nil
+	case "pid":
+		return rspec.Namespace{Type: rspec.PIDNamespace, Path: path}, nil
+	case "mount":
+		return rspec.Namespace{Type: rspec.MountNamespace, Path: path}, nil
+	case "ipc":
+		return rspec.Namespace{Type: rspec.IPCNamespace, Path: path}, nil
+	case "uts":
+		return rspec.Namespace{Type: rspec.UTSNamespace, Path: path}, nil
+	case "user":
+		return rspec.Namespace{Type: rspec.UserNamespace, Path: path}, nil
+	case "cgroup":
+		return rspec.Namespace{Type: rspec.CgroupNamespace, Path: path}, nil
+	default:
+		return rspec.Namespace{}, fmt.Errorf("Should not reach here!")
+	}
+}
+
+// SetLinuxNamespace sets g.spec.Linux.Namespaces.
+func (g Generator) SetLinuxNamespace(nsMaps map[string]string) error {
+	g.spec.Linux.Namespaces = []rspec.Namespace{}
+	for ns, path := range nsMaps {
+		if err := g.AddOrReplaceLinuxNamespace(ns, path); err != nil {
+			return nil
+		}
+	}
+	return nil
+}
+
+// AddOrReplaceLinuxNamespace adds or replaces a namespace inside
+// g.spec.Linux.Namespaces.
+func (g Generator) AddOrReplaceLinuxNamespace(ns string, path string) error {
+	namespace, err := mapStrToNamespace(ns, path)
+	if err != nil {
+		return err
+	}
+
+	for i, ns := range g.spec.Linux.Namespaces {
+		if ns.Type == namespace.Type {
+			g.spec.Linux.Namespaces[i] = namespace
+			return nil
+		}
+	}
+	g.spec.Linux.Namespaces = append(g.spec.Linux.Namespaces, namespace)
+	return nil
+}
+
+// RemoveLinuxNamespace removes a namespace from g.spec.Linux.Namespaces.
+func (g Generator) RemoveLinuxNamespace(ns string) error {
+	namespace, err := mapStrToNamespace(ns, "")
+	if err != nil {
+		return err
+	}
+
+	for i, ns := range g.spec.Linux.Namespaces {
+		if ns.Type == namespace.Type {
+			g.spec.Linux.Namespaces = append(g.spec.Linux.Namespaces[:i], g.spec.Linux.Namespaces[i+1:]...)
+			return nil
+		}
+	}
+	return nil
+}
+
+// StrPtr returns the pointer pointing to the string s.
+func StrPtr(s string) *string { return &s }
```

Comment:
I may remember the wrong variable. `strPtr` is only used by the generate library and can be private.